scipy/io/mmio.py

```diff
@@ -281,20 +286,44 @@ def _get_symmetry(a):
         m,n = a.shape
         if m != n:
             return MMFile.SYMMETRY_GENERAL
-        issymm = 1
-        isskew = 1
+        issymm = True
+        isskew = True
         isherm = a.dtype.char in 'FD'
-        for j in range(n):
-            for i in range(j+1,n):
-                aij,aji = a[i][j],a[j][i]
-                if issymm and aij != aji:
-                    issymm = 0
-                if isskew and aij != -aji:
-                    isskew = 0
-                if isherm and aij != conj(aji):
-                    isherm = 0
-                if not (issymm or isskew or isherm):
-                    break
+        
+        if isinstance(a,spmatrix):
+            # check if number of nonzero entrys of lower and upper triangle matrix are equal
+            (row, col) = a.nonzero()
+            if (row < col).sum() != (row > col).sum():
+                return MMFile.SYMMETRY_GENERAL
+            
```

Comment:
Also, the proposed additional symmetry test considers only the sparsity pattern of the matrix. In order to compare the values of the matrix the following test has to be inserted too. Then the element wise compare would be eliminate.
```
# check data for symmetry
data_transpose = data[sort_rows]
data_transpose = data_transpose[sort_cols]

if np.all(data == data_transpose):
    return MMFile.SYMMETRY_SYMMETRIC

if np.all(data == - data_transpose):
    return MMFile.SYMMETRY_SKEW_SYMMETRIC
    
if np.all(data == conj(data_transpose)):
    return MMFile.SYMMETRY_HERMITIAN

return MMFile.SYMMETRY_GENERAL
```